frontend/components/MainContent/MainContent.tsx

```diff
@@ -80,10 +84,16 @@ const MainContent = ({
 
     return null;
   };
+
+  const appWideBanner = renderAppWideBanner();
+  const mainContentConfig: IMainContentConfig = {
+    renderedBanner: !!appWideBanner,
+  };
+
   return (
     <div className={classes}>
-      {renderAppWideBanner()}
-      {children}
+      {appWideBanner}
+      {typeof children === "function" ? children(mainContentConfig) : children}
```

Comment:
👀

Reply:
it looks a little uggo to me but I couldn't think of a more React-y way of doing it 🤷 .  Looks like we have a similar pattern in a couple other places.